pkg/tls/options.go

```diff
@@ -0,0 +1,102 @@
+package tls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+)
+
+func DefaultServerOpts(logger log.Logger, cert, key, clientCA string) (*tls.Config, error) {
```

Comment:
Nit, now that this is changing places and context, I think it would make sense to rename it something like:
`NewServerConfig`